crates/syntax/src/validation.rs

```diff
@@ -231,7 +231,7 @@ fn validate_visibility(vis: ast::Visibility, errors: &mut Vec<SyntaxError>) {
         Some(it) => it,
         None => return,
     };
-    if impl_def.trait_().is_some() {
+    if impl_def.trait_().is_some() && impl_def.attrs().next().is_none() {
```

Comment:
Can you add a FIXME comment documenting that this is a workaround? Something like this:

```suggestion
    // FIXME: disable validation if there's an attribute, since some proc macros use this syntax.
    // ideally the validation would run only on the fully expanded code, then this wouldn't be necessary.
    if impl_def.trait_().is_some() && impl_def.attrs().next().is_none() {
```

Reply:
Done